libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp

```diff
@@ -0,0 +1,328 @@
+            new_throttle = (uint16_t)SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
+
+            // Limit throttle percent to 65% for testing purposes only
+            if (new_throttle > 65)
+            {
+                new_throttle = 65;
+            }
+
+            //check for change or timeout for throttle value
+            if ((new_throttle != old_throttle) || (now - last_req_send_throttle > 500)){
+                // if new throttle value, send throttle request
+                status = send_target_values(new_throttle);
+                old_throttle = new_throttle;
+                internal_state.k_throttle = new_throttle;
+                last_req_send_throttle = now;
+            }
+            else {
+                //request Status request, if no throttle commands
+                status = send_request_status();
+            }      
+
+            if (status == true) {
+                waiting_response = true;
+                last_response_ms = now;
+            }
+        }
+    }
+}
+
+
+/**
+ * @brief updates the current quantity that will be expected
+ * 
+ */
+void AP_EFI_Serial_Hirth::get_quantity() {
+    switch (req_data.code)
+    {
+    case CODE_REQUEST_STATUS_1:
+        expected_bytes = QUANTITY_RESPONSE_STATUS_1;
+        break;
+    case CODE_REQUEST_STATUS_2:
+        expected_bytes = QUANTITY_RESPONSE_STATUS_2;
+        break;
+    case CODE_REQUEST_STATUS_3:
+        expected_bytes = QUANTITY_RESPONSE_STATUS_3;
+        break;
+    case CODE_SET_VALUE:
+        expected_bytes = QUANTITY_ACK_SET_VALUES;
+        break;
+    }
+}
+
+
+/**
+ * @brief sends the new throttle command to Hirth ECU
+ * 
+ * @param thr - new throttle value given by SRV_Channel::k_throttle
+ * @return true - if success
+ * @return false - currently not implemented
+ */
+bool AP_EFI_Serial_Hirth::send_target_values(uint16_t thr) {
+    int idx = 0;
+    uint8_t computed_checksum = 0;
+
+    //clear buffer
+    for (size_t i = 0; i < sizeof(raw_data); i++)
+    {
+        raw_data[i] = 0;
+    }
+    
+    // Get throttle value
+    uint16_t throttle = thr * THROTTLE_POSITION_FACTOR;
+
+    // set Quantity + Code + "20 bytes of records to set" + Checksum
+    computed_checksum += raw_data[idx++] = req_data.quantity = QUANTITY_SET_VALUE;
+    computed_checksum += raw_data[idx++] = req_data.code = CODE_SET_VALUE;
+    computed_checksum += raw_data[idx++] = throttle & 0xFF;
+    computed_checksum += raw_data[idx++] = (throttle >> 0x08) & 0xFF;
+
+    for (; idx < QUANTITY_SET_VALUE - 2; idx++) {
+        // 0 has no impact on checksum
+        raw_data[idx] = 0;
+    }
+    //checksum calculation
+    computed_checksum = computed_checksum % 256;
+    raw_data[QUANTITY_SET_VALUE - 1] = (256 - computed_checksum);
+
+    //debug
+    internal_state.packet_sent = 5;
+    data_send =5;
+    
+    expected_bytes = QUANTITY_ACK_SET_VALUES;
+    //write data
+    port->write(raw_data, QUANTITY_SET_VALUE);
```

Comment:
Should check `port` has enough space for this data before writing it.

Reply:
Will need your help with this. I could not find a supporting method available in UARTDriver.cpp